akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala

```diff
@@ -706,23 +713,43 @@ class TestKit(system: ActorSystem) {
    * @return the last received message, i.e. the first one for which the
    *         partial function returned true
    */
-  def fishForMessage(max: Duration, hint: String, f: JFunction[Any, Boolean]): Any = {
+  @deprecated("Use the overloaded one which accepts java.time.Duration instead.", since = "2.6.0")
+  def fishForMessage(max: Duration, hint: String, f: JFunction[Any, Boolean]): Any =
     tp.fishForMessage(max, hint)(new CachingPartialFunction[Any, Boolean] {
       @throws(classOf[Exception])
       override def `match`(x: Any): Boolean = f.apply(x)
     })
-  }
+
+  /**
+   * Hybrid of expectMsgPF and receiveWhile: receive messages while the
+   * partial function matches and returns false. Use it to ignore certain
+   * messages while waiting for a specific message.
+   *
+   * @return the last received message, i.e. the first one for which the
+   *         partial function returned true
+   */
+  @silent("deprecated")
+  def fishForMessage(max: java.time.Duration, hint: String, f: JFunction[Any, Boolean]): Any =
+    fishForMessage(max.asScala, hint, f)
 
   /**
    * Same as `fishForMessage`, but gets a different partial function and returns properly typed message.
    */
+  @deprecated("Use the overloaded one which accepts java.time.Duration instead.", since = "2.6.0")
   def fishForSpecificMessage[T](max: Duration, hint: String, f: JFunction[Any, T]): T = {
     tp.fishForSpecificMessage(max, hint)(new CachingPartialFunction[Any, T] {
       @throws(classOf[Exception])
       override def `match`(x: Any): T = f.apply(x)
     })
   }
 
+  /**
+   * Same as `fishForMessage`, but gets a different partial function and returns properly typed message.
+   */
+  @silent("deprecated")
+  def fishForSpecificMessage[T](max: java.time.Duration, hint: String, f: JFunction[Any, T]): T =
```

Comment:
I must say that I don't  understand how I would use this based on the doc comment. What `T`  should the function return?  How does it say that a message is matching and not?

Reply:
Yeah, I agree I just filled in a missing variation here to make it consistent. Noted and created #27562 to track fixing those previous copy-pasta probably useless methods saying PF but taking a total function in Java.

Reply:
I  think it's more problematic than that. This API makes no sense, how would you say "not matching" when it's not a partial function. I would be ok with just removing fishForSpecificMessage in javadsl.

Reply:
Argh, there's even Java sample docs for this, I'll revert removing these and still leave making the API reasonable to a later point. Wouldn't want to remove them without a replacement.